crates/broker/src/config.rs

```diff
@@ -405,6 +420,36 @@ impl ConfigWatcher {
     }
 }
 
+// Helper module so we can do #[serde(with = "hex_opt_vec")] for optional hex encoded Vec<u8>
+mod hex_opt_vec {
+    use super::*;
+    use hex::serde as hex_serde;
+
+    pub fn serialize<S>(value: &Option<Vec<u8>>, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: serde::Serializer,
+    {
+        #[derive(Serialize)]
+        struct Helper<'a>(#[serde(with = "hex_serde")] &'a Vec<u8>);
+
+        match value {
+            Some(v) => Helper(v).serialize(serializer),
+            None => serializer.serialize_none(),
+        }
+    }
+
+    pub fn deserialize<'de, D>(deserializer: D) -> Result<Option<Vec<u8>>, D::Error>
+    where
+        D: serde::Deserializer<'de>,
+    {
+        #[derive(Deserialize)]
+        struct Helper(#[serde(with = "hex_serde")] Vec<u8>);
+
+        let helper = Option::deserialize(deserializer)?;
+        Ok(helper.map(|Helper(v)| v))
```

Comment:
This may work for toml/json, just noting that this will not work for some serialization protocols. Probably won't ever matter in our case, but wondering why handling `Option` case implicitly. `serde_with` crate would remove the need for all of this.